channeldb/invoices.go

```diff
@@ -1882,9 +1887,14 @@ func updateInvoiceState(invoice *Invoice, hash lntypes.Hash,
 	// or canceled. The only restriction is on transitioning to settled
 	// where we ensure the preimage is valid.
 	case ContractOpen:
-		if update.NewState == ContractSettled {
-			// Validate preimage.
+		if update.NewState != ContractSettled {
+			invoice.State = update.NewState
```

Comment:
Is it better to not use the catch all state set at the end of this function anywhere?

Reply:
to be clear, you're suggesting removing the catch all and making each transition explicit? i do prefer that

Reply:
yes, but nothing important.

Reply:
catch all is still there, but I think the control flow is much clearer with the helper extracted